fdbclient/NativeAPI.actor.cpp

```diff
@@ -4872,48 +4886,97 @@ Future<Standalone<StringRef>> Transaction::getVersionstamp() {
 	return versionstampPromise.getFuture();
 }
 
-ACTOR Future<ProtocolVersion> coordinatorProtocolsFetcher(Reference<ClusterConnectionFile> f) {
-	state ClientCoordinators coord(f);
+// Gets the protocol version reported by a coordinator via the protocol info interface
+ACTOR Future<ProtocolVersion> getCoordinatorProtocol(NetworkAddressList coordinatorAddresses) {
+	RequestStream<ProtocolInfoRequest> requestStream{ Endpoint{ { coordinatorAddresses }, WLTOKEN_PROTOCOL_INFO } };
```

Comment:
Should this endpoint be included into the `ClientLeaderRegInterface`?

Reply:
It's a good question. I think probably it's out of scope of this PR, but it's worth thinking about whether there's a better way to expose this endpoint.

Possibly we could do something like your suggestion, though that may not be the most natural place for it given this endpoint is available on all workers. You could do something like the `WorkerInterface` maybe, though in that case I'd have to see if it's awkward to get a `WorkerInterface` for a process that we haven't connected to and may not be compatible with.

Reply:
It's used in `getCoordinatorProtocol` and I thought hmm that endpoint should be part of the interface that a client process uses to interact with Coordinators, which in this case is `ClientLeaderRegInterface`. Why isn't it the most natural place?

Reply:
Every worker exposes this interface, and we just happen to be getting it from a coordinator in this case.

Reply:
Ah, I see. It is indeed tricky to find a good place for it.